### DISCUSSION TITLE
How to incorporate local system configs into Salt?

### DISCUSSION THREAD

**Discussion by abesinger** [Security-Onion-Solutions/securityonion#6943] [Unsupported Versions]
I've got some system configurations that I've done, such as added shell users, installed some packages, configured sshd.conf, etc., that I'd like to make Salty so that they automatically get applied to all new nodes that I add to my cluster.

Is there a place in SO's Salt configuration where I  can put local salt/pillars that won't get overwritten when I run SOUP, nor override a `saltstack/default` file with one in `saltstack/local`?

**Comment by petiepooo** [comment]
I was using salt for remote sensor orchestration before SO2 took over and depended on being salt-master.  I switched to using salt-ssh so the sensors were agentless minions.  It's not perfect, but it allows me to manage a bunch of standalone SO2 deployments from one master salt server.  Before you ask, they're separate by design and cannot be in the same grid.

**Reply by abesinger** [reply]
thanks, but that doesn't really address my question.  I *want* to use SO as the salt-master.  To use `salt` parlance, I want to be able to define additional states that get applied across SO hosts, for example, adding a Linux user account. I'd essentially like something get run as part of `/opt/so/saltstack/default/salt/common/init.sls`, without having to hack in something that can get displaced with SOUP.

**Comment by abesinger** [comment]
Looking at `/opt/so/saltstack/default/salt/common/init.sls`, I think what I'm looking for is an addition such as:

  {% do allowed_states.append('local_site') %}

With a dummy file in `/opt/so/saltstack/default/salt/local_site` that does nothing, that I could override with one in `/opt/so/saltstack/local/salt/local_site`

**Reply by petiepooo** [reply]
The recommendation is to not change anything under default as that belongs to soup and could get overwritten on update. What I have done is copy files out of default into the same location but under local. They are less likely to be overwritten on update, but you would still have to merge any changes they made to the default files into your local copy. I recommend you keep track of the original file as it was before editing to make it easier to track changes. Git can help with that.

**Comment by weslambert** [comment]
As mentioned, the recommendation is put your things in the equivalent `/opt/so/saltstack/local` directory. SOUP _should_ warn you if there are files in the `local` directory that are modified from the default. Custom files, however, _should_ be left untouched.

**Reply by abesinger** [reply]
@weslambert You've both missed the point of my question, which is asking whether the Salt config already as something in it which would allow me to add Salt states _without_ having to maintain a modified copy of one of the default files and keep it in `/opt/so/saltstack/local`. I'm trying to avoid doing just that.

I'm looking for the analog of `rc.local` where I can drop in local configurations without modifying the system files., I don't want to have to maintain my configs by by hand, nor have to set up some separate configuration management infrastructure.

In my comment previous to this one, I was suggesting how it might be done, but I don't want to presume that it's not already in the Salt config, perhaps I've missed something...
